lib/setupHooks/installFromCargoBuildLogHook.sh

```diff
@@ -13,12 +13,18 @@ function installFromCargoBuildLog() (
   logs=$(jq -R 'fromjson?' <"${log}")
 
   local select_non_test='select(.reason == "compiler-artifact" and .profile.test == false)'
-  local select_bins="${select_non_test} | .executable | select(.!= null)"
+  local select_non_dep='select((contains("/release/deps/") or contains("/debug/deps/"))| not)'
+  local select_bins="${select_non_test}"'| .executable | select(.!= null) | '"${select_non_dep}"
   local select_lib_files="${select_non_test}"'
-    | select(.target.kind | contains(["staticlib"]) or contains(["cdylib"]))
+    | select(.target.kind
+        | contains(["cdylib"])
+        or contains(["dylib"])
+        or contains(["lib"])
+        or contains(["rlib"])
```

Comment:
Question: why do we want to install `rlibs` at all? These are intermediate artifacts used by the compiler and are generally (to my knowledge) not very useful outside of that. Plus the installed cargo artifacts dir should already include these (and in fact installing them all would largely result in installing the contents of cargo's `target` directory IIUC). I'd imagine if someone really needs those artifacts they should probably look for them there and not inside of `$out/lib`

Reply:
SGTM